test/cases/schema_statements_test.rb

```diff
@@ -5,22 +5,25 @@
 module PostGIS
   class SchemaStatementsTest < ActiveSupport::TestCase
     def test_initialize_type_map
-      initialized_types = SpatialModel.connection.send(:type_map).keys
+      SpatialModel.with_connection do |connection|
+        assert connection.connected?
```

Comment:
This is flacky.

Reply:
Probably due to how the adapter is registered now?

Reply:
No, it just flacky in CI. 

I will focus on the cause later.

Reply:
Was it flaky before this PR?

Reply:
Yes, in CI.